### ISSUE TITLE
Various issues with landing rockets

### ISSUE THREAD

**Issue by EzerArch** [micdoodle8/Galacticraft#2459]
_(I created the issue to investigate [this bug report on the forum](https://forum.micdoodle8.com/index.php?threads/space-station-in-release-1-7-3-0-12-4xx.6638/#post-42888), but I ended up finding this one)_

I've set up 2 LCs, one on the OW (frequency 1), and other at the space station (frequency 2).

Both LCs can detect each other, but instead of performing the self-landing procedure, the mod opens the galaxy map and you have to spawn in the dimension - at the SS arrival block or with parachute, according to the target dim.

**Video (sped up 1.25x):**  https://youtu.be/LUodHDezyTE

No other mod installed but GC 477 and NEI, and my resource pack.

**Comment by radfast** [comment]
This should be fixed in GC 478.

At the same time, while testing I noticed and fixed a really nasty, long-standing bug.  (Nature of that bug:  either a (non-Galacticraft, vanilla code) crash if launch rocket to space from a space station if there are **no** other entities on the space station, otherwise **random deletion** of one or two entities in the world where the rocket was launched from - hopefully chickens or squid most of the time, but sometimes it could have been rockets waiting on launchpads or other valuable entities.  This most likely explains random reports of "I lost my rocket" - especially in Space Station dimensions where the only other entity in the dimension might be a rocket!)

-----

Some rocketry issues I'm still seeing, will try to fix later this week:
1.  Occasionally a rocket in landing mode "reappears" when I relog onto a world, even if it already landed on the pad before I quit and saved the world (this gives possible duplication of the rocket)
2.  If a player starts the launch 20 second countdown, lets it reach a small number like 3 seconds, then relog, the countdown starts again at 20 seconds
3.  Rocket in landing mode sometimes visually judders on client, blipping up or down from its correct position - worse when velocity is high.
4.  Cargo rockets have problems landing when another rocket is already on the target pad - the intended behaviour is that the rocket lands, and breaks and drops as an item any rocket which is already present (you can think of it as being damaged by the flames of the descending rocket).  Intended behaviour is working OK with passenger rockets I think
5.  If the player attacks and breaks a rocket which is in flight - easiest to do this for a landing rocket, while it is moving slowly just before it lands on a pad - then it drops as an item twice, giving duplication.  If the rocket is on a landing pad, attacking it drops only 1 rocket item which is the intended behaviour.

If you notice any of these happening when testing 478, I'd be happy to have that confirmation from a second person :)   Not sure if these are long-standing rare issues or recently introduced.

I will leave this issue open as a reminder to me.

**Comment by radfast** [comment]
I don't think this is related to the space station creation issue reported in the forum, so that will need a separate issue I guess ...  The space station creation issue could be the same as #2047

**Comment by EzerArch** [comment]
> so that will need a separate issue I guess

I'll post it when I get some spare time. I need to find out which mod is causing the issue, once I have no problems with creating the SS if it's Galacticraft alone (as seen here https://youtu.be/LUodHDezyTE?t=1m36s)

**Comment by radfast** [comment]
I really appreciate your help with testing those types of issues :)
We are getting there, I am currently working through the 1 to 5 which I posted
Hopefully soon a "final" version of GC 1.7.10 which is free of major issues, then all efforts can be focused on GC4.

**Comment by EzerArch** [comment]
Off-topic: I have good news. The Optifine's IndexOutOfBoundsException that I usually get when I ride a rocket from SS to any other dim is gone. The bad news is I don't know whom I should credit it to: I updated both GC (with your fixes) and Optifine these days. :D

**Comment by EzerArch** [comment]
> 1. Occasionally a rocket in landing mode "reappears" when I relog onto a world, even if it already landed on the pad before I quit and saved the world (this gives possible duplication of the rocket)

Confirmed. At 5:00 you'll see rocket dupe happening in the other dims: https://youtu.be/GRpGLusVp28 (uploading)

**Comment by radfast** [comment]
1, 2, and 4 now fixed.  Number 5 seems to be an issue in 1.8.9 only (and may also be fixed by the changes I just made to 1.7.10)

The visual judder when a rocket is landing I'm finding difficult to fix, maybe @micdoodle8 will have some ideas?  It's connected with network latency and the client receiving position updates from the server.
